### PR TITLE
implement queries for GET /api/articles

### PR DESCRIPTION
**Pull request by dmp990**

### CODE REVIEW

File: error_handlers/errors.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ exports.handleCustomErrors = (err, req, res, next) => {
   }
 };
 
-exports.unhandledErrors = (req, res) => {
+exports.unhandledErrors = (err, req, res) => {
+  console.log("here")
```
Comment (CatHoang) [inline error_handlers/errors.js:14]: Try and delete console logs when you have finished with them

File: models/articlesModels.js [hunk 2]
```diff
@@ -20,7 +21,10 @@ exports.fetchArticleById = ({ article_id }) => {
       if (result.rows.length) {
         return result.rows[0];
       }
-      return Promise.reject({ status: 404, msg: "article not found" });
+      return Promise.reject({
+        status: 404,
+        msg: "article not found",
+      });
     })
     .catch((err) => {
```
Comment (CatHoang) [inline models/articlesModels.js:29]: we would want to catch all our errors in either the controller or models not both

File: models/articlesModels.js [hunk 3]
```diff
@@ -55,26 +59,55 @@ exports.updateArticleById = ({ article_id }, { inc_votes }) => {
...
-        GROUP BY articles.article_id
-        ORDER BY articles.created_at DESC;`
-    )
-    .then((result) => {
-      return result.rows;
+    .query(queryStr)
+    .then(({ rows }) => {
+      return rows;
     })
     .catch((err) => {
```
Comment (CatHoang) [inline models/articlesModels.js:109]: we should do our catch either in our controller or models and stick with one.

**Review by CatHoang — COMMENTED** [review]
Hi Asad, have made some minor comments for you to look at and review. Looking really good :)

File: models/articlesModels.js (hunk 3 above)
Comment (dmp990) [reply models/articlesModels.js:109]: So if `db.query` returns an error and I don't have a `catch` block in my models, will the `catch` in my controllers be able to catch it?

**Review by CatHoang — APPROVED** [review]
Hi Asad, looks great - feel free to merge :)

**Pull request merged**